x/stakeibc/keeper/community_pool.go

```diff
@@ -1,12 +1,104 @@
+	"errors"
+	"fmt"
+
+	sdkmath "cosmossdk.io/math"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/Stride-Labs/stride/v14/utils"
 )
 
+// For each hostZone with a valid community pool, trigger the ICQs and ICAs to transfer deposited tokens to holding address
+// Since ICQs and ICAs take time to complete, it is almost certain tokens swept in and processed will be swept out in a later epoch
+func (k Keeper) SweepInAllDepositedCommunityPoolTokens(ctx sdk.Context) error {
+	hostZones := k.GetAllActiveHostZone(ctx)
+	for _, hostZone := range hostZones {
+		if hostZone.CommunityPoolDepositIcaAddress != "" &&
+			hostZone.CommunityPoolHoldingAddress != "" &&
+			hostZone.CommunityPoolReturnIcaAddress != "" {
+				k.ICQCommunityPoolDepositICABalances(ctx, hostZone.ChainId)
+			}
+	}
+	return nil
+}
+
+// For each hostZone with a valid community pool, trigger the transfers of tokens in the holding address to the return ICA
+func (k Keeper) SweepOutAllReturningCommunityPoolTokens(ctx sdk.Context) error {
+	hostZones := k.GetAllActiveHostZone(ctx)
+	for _, hostZone := range hostZones {
+		if hostZone.CommunityPoolDepositIcaAddress != "" &&
+			hostZone.CommunityPoolHoldingAddress != "" &&
+			hostZone.CommunityPoolReturnIcaAddress != "" {
+				k.IBCReturnAllCommunityPoolTokens(ctx, hostZone)
+			}
+	}	
+	return nil
+}
+
+
+
+// Place holder function, will ICQ to see what balances exist in the deposit ICA on the community pool host zone
+// In the ICQ callback, for existing balances it will call IBCTransferCommunityPoolICATokensToStride with found token as input
+func (k Keeper) ICQCommunityPoolDepositICABalances(ctx sdk.Context, communityPoolHostZoneId string) error {
+	return nil
+}
+
+
 // ibc transfers tokens from the foreign hub community pool deposit ICA address onto Stride hub
-func (k Keeper) IBCTransferCommunityPoolICATokensToStride(ctx sdk.Context, token sdk.Coin) error {
-	k.Logger(ctx).Info("Transfering tokens from community pool to Stride...")
+// then as an atomic action, liquid stake the tokens with Autopilot commands in the ibc message memos
+func (k Keeper) IBCTransferCommunityPoolICATokensToStride(ctx sdk.Context, communityPoolHostZoneId string, token sdk.Coin) error {
+	k.Logger(ctx).Info(fmt.Sprintf("Transfering %d %s tokens from community pool deposit ICA to Stride hub holding address", token.Amount.Int64(), token.Denom))
+
+	// TODO: add safety check here on if the amount is greater than a threshold to avoid many, frequent, small transfers
+	//       threshold might be a config we tune to avoid ddos type attacks, for now using 0 as hard coded threshold
+	if token.Amount.LTE(sdkmath.ZeroInt()) {
+		k.Logger(ctx).Error(fmt.Sprintf("[IBCTransferCommunityPoolICATokensToStride] The amount %v to transfer did not meet the minimum threshold!", token.Amount))
+		return errors.New("Transfer Amount below threshold!")
+	}
+
+	hostZone, hostZoneFound := k.GetHostZone(ctx, communityPoolHostZoneId)
+	if !hostZoneFound || hostZone.Halted {
```

Comment:
impressive catch making sure to check the hz isn't halted here!

(side note - out of scope of this PR) I wonder if we should add a function `GetActiveHostZone` like
```go
func (k Keeper) GetActiveHostZone(ctx sdk.Context, hostZoneId string) (types.HostZone, error) {
    hostZone, found := k.GetHostZone(ctx, hostZoneId) 
    if !found {
       return types.ErrHostNotFound.Wrap(hostZoneId)
    }
    if host.Halted {
       return types.ErrHostHalted.Wrap(hostZoneId)
    }
    return nil, hostZone
}
```

That way, we don't have to (a) duplicate the hostNotFound error that we have in every function and (b) worry about forgetting to check if a host is halted

Reply:
Yes, we do this pattern everywhere and not terribly consistently.  This suggested function is a good idea in a lot of places in the codebase, maybe if we ever refactor stakeibc like I want to this would be a useful helper since it normally needs to be checked when going from an id to a hostzone variable often in callbacks